notifications/models.py

```diff
@@ -1,13 +1,43 @@
+import requests
 from django.db import models
-
+from django.db.models import Max
+from django.contrib.postgres.fields import JSONField, ArrayField
 from django.contrib.auth.models import User
-from councilmatic_core.models import Bill, Organization, Person
+from django.conf import settings
+from django.utils import timezone
+from councilmatic_core import models as councilmatic_models
+from opencivicdata.core.models.base import OCDIDField
 
-from django.contrib.postgres.fields import JSONField
+try:
+    HAYSTACK_URL = settings.HAYSTACK_CONNECTIONS['default']['URL']
+except KeyError:
```

Comment:
this is totally up to your judgment: do you think we should log a warning that solr is not configured if the app starts without this setting, either in addition to or instead of logging when the management command is sent?

Reply:
I like that idea a lot, but do you have any sense of how we'd hook into the app start event to emit the warning? Not sure when that happens and I can't find anything from a cursory search.

Reply:
i think that this will get evaluated when you start the app, no extra steps required. (added a print statement to the except block to test.)

```
(chi-councilmatic) call-me-hank:chi-councilmatic hannah$ python manage.py runserver
no haystack url
```

that said, as i was testing that assumption, i learned haystack itself will raise a fatal exception if you don't define the default connection, so maybe we don't need to do anything ourselves.

```
    raise ImproperlyConfigured("The default alias '%s' must be included in the HAYSTACK_CONNECTIONS setting." % DEFAULT_ALIAS)
django.core.exceptions.ImproperlyConfigured: The default alias 'default' must be included in the HAYSTACK_CONNECTIONS setting.
```

what do you think?

Reply:
Nice! I think the built-in error is definitely more informative than ours anyway. The only open question to me is whether or not there's a use-case for running this app without search support. If so, it might make more sense to handle the error gently with a warning rather than raise a fatal error IMO. But if we imagine that wiring up Haystack is a necessary prerequisite for the app, then I think we can safely get rid of our error and rely on Haystack to complain.

Reply:
haystack is a pre-req of django-councilmatic, so if we can't envision running notifications without django-councilmatic, then i think haystack is a necessary pre-req here. does that logic check out?

Reply:
Makes sense to me 👍

Reply:
i think we we landed on the app not being able to run without search, i.e., we can rely on haystack to raise the configuration errors.

is there any reason to keep the error handling in:

- this check?
- `BillSearchSubscription.get_updates`?
- `send_notifications`?

(apologies if this is obvious, i may need some more caffeine.)

Reply:
To me, these extra checks seem like reasonable defensive programming in the face of an uncertain integration. We expect Haystack to raise a configuration error if its configuration dictionary is not properly defined, but we also don't control Haystack or its startup logic, so we can't have perfect confidence that it will error out when we want it to. Plus, when we access `HAYSTACK_URL`, it's in a nested dictionary on the `settings` object, so if any part of `settings.HAYSTACK_CONNECTIONS['default']['URL']` is not defined the user will get a super unhlpeful error. Does this judgment seem reasonable to you, or do you think it's overly confusing to be extra cautious around `HAYSTACK_URL`?

Reply:
while it's true we don't control haystack, we do control the version of haystack we want to run, so we have a way of ensuring consistency if future releases of haystack behave differently than the one we're using now, which does raise the configuration error as we expect.

that said, i do find more useful exceptions a compelling reason to preserve the custom exception handling code. this isn't a change i require.

Reply:
Cool 👍